ui/app/scripts/controllers/serviceVersionStatusCtrl.js

```diff
@@ -214,55 +181,53 @@ angular.module('apollo')
                 }
             };
 
+            $scope.setWebsocketScope = function (scope) {
+                $scope.websocketScope = scope;
+
+                if (scope === execTypeName) {
+                    $scope.terminalHeader = "Live Session to " + $scope.selectedPodStatus.name;
+                } else if (scope === logsTypeName) {
+                    $scope.terminalHeader = "Live Tail on " + $scope.selectedPodStatus.name;
+                }
+            };
+
             $scope.openHawtio = function (podStatus) {
                 $window.open(apolloApiService.getHawtioLink($scope.selectedStatus.environmentId, podStatus.name));
             };
 
-            function refreshPreSelectedStatus() {
-                if ($scope.selectedStatus) {
-
-                    // Refresh the view for the user
-                    $scope.selectedStatus = $scope.filteredResults.filter(function (filtered) {
-                        if (filtered) {
-                            return (filtered.serviceId === $scope.selectedStatus.serviceId && filtered.environmentId === $scope.selectedStatus.environmentId);
-                        }
-                    })[0];
+            function populateDeployableVersions() {
+                $scope.deployableVersion;
+                if ($scope.kubernetesDeploymentStatus !== null) {
+                    apolloApiService.getDeployableVersionBasedOnSha($scope.kubernetesDeploymentStatus.gitCommitSha, $scope.kubernetesDeploymentStatus.serviceId).then(function (response) {
```

Comment:
kubernetesDeploymentStatus is an array, you can't just do .gitCommitSha and .serviceId... Have you tested it?